### ISSUE TITLE
Asciidoctor 2.0.0 does not output the stack trace

### ISSUE THREAD

**Issue by Mogztter** [asciidoctor/asciidoctor.js#699]
When an error (exception) occurred, Asciidoctor 2.0.0 does not output the stack trace. For instance:

```
asciidoctor: FAILED: index.adoc: Failed to load AsciiDoc document - undefined method `start_with?' for nil
```

I don't know if it's intended or not but it's very hard to debug.
We should at least document how to enable the stack trace (if it's possible).

**Comment by mojavelinux** [comment]
It should wrapping the original exception. If it's not, that's an indication that something in Opal either changed or it was never supported.

Here's the wrapping logic: https://github.com/asciidoctor/asciidoctor/blob/bf53c82342e73b12e06093a83fb1ff0c9c14f920/lib/asciidoctor.rb#L1305-L1322

**Comment by Mogztter** [comment]
You're right: https://github.com/opal/opal/issues/1962

**Comment by mojavelinux** [comment]
If there's something we can do to workaround it in core in the meantime, feel free to suggest a patch.

**Comment by Mogztter** [comment]
> If there's something we can do to workaround it in core in the meantime, feel free to suggest a patch.

@mojavelinux  There's at least one workaround:

```rb
`wrapped_ex.stack = ex.stack` if RUBY_ENGINE == 'opal'
```

In Opal, Ruby `Exception` inherit from JavaScript `Error`. So we can assign the `stack` property.
But maybe it's "dangerous" to use the backtick operator in here ?

**Comment by mojavelinux** [comment]
If the exception responds to `stack=`, I think we can use that instead of `stack_trace=`. Though maybe that's just the solution for Opal in the first place.

**Comment by Mogztter** [comment]
I can confirm that the following code does the trick:

```rb
`wrapped_ex.stack = ex.stack` if RUBY_ENGINE == 'opal'
```

> if the exception responds to stack=

The exception does not respond to `stack=`. The stack property is a native JavaScript property on the `Error` object: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Stack
That's why I'm using quoted text in order to execute plain JavaScript code.

> Though maybe that's just the solution for Opal in the first place.

It's not a critical issue so I think we should wait for upstream fix: opal/opal#1962

**Comment by mojavelinux** [comment]
Ah, gotcha. Thanks for clarifying.
